# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/pr120107/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [6]:
import os
import sys
import subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(
            ["git", "clone", "--depth", "1", REPO_URL, REPO_DIR],
            check=True
        )

    os.chdir(REPO_DIR)

    subprocess.run(
        [sys.executable, "-m", "pip", "install", "-q", "-r", "requirements.txt"],
        check=True
    )

print("Working directory:", os.getcwd())

Working directory: /content/flyrank-ml-internship-starter


# My Lane: Structured Content Archetype Clustering

I chose the **Structured Content Archetype Clustering** lane because I want to identify different types of content pages based on their performance and characteristics. Instead of predicting a single outcome, clustering can reveal natural groups of pages that behave similarly. These groups can help content teams understand their content inventory and decide whether pages should be protected, improved, refreshed, monitored, or pruned.

# The Question

### Decision

How can we group content pages into meaningful performance archetypes to support content management decisions?

### Unit of Analysis

One content page.

### Output

A set of content archetypes (clusters), each described by its common characteristics and suggested action.

### Action

Content reviewers can use these archetypes to decide whether a group of pages should be protected, refreshed, improved, merged, monitored, or pruned.

### Cost of a Wrong Recommendation

If pages are assigned to an unsuitable archetype, reviewers may spend time on the wrong content or overlook pages that need attention. The clustering is intended to support human decision-making rather than replace it.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Content types:", df["content_type"].nunique())
print("Average CTR:", round(df["ctr"].mean(), 3))
print("Average content age (days):", round(df["content_age_days"].mean(), 1))

Total pages: 30000
Content types: 3
Average CTR: 0.511
Average content age (days): 256.2


The starter dataset contains **30,000 pages**, which provides enough data for clustering.

There are **3 content types**, allowing different kinds of pages to be compared.

The average CTR is **0.511** and the average content age is **256.2 days**, indicating variation in engagement and content maturity. These measured values suggest the dataset is suitable for identifying meaningful content archetypes.

## What I can and can't claim

This project identifies groups of pages with similar characteristics based on the available data.

The clusters can support content review decisions, but they do not prove that one factor causes better or worse performance.

The results are observational and intended to help prioritize content decisions rather than predict Google's ranking algorithm or guarantee traffic improvements.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.